crypto/ec/ec_curve.c

```diff
@@ -3191,9 +3274,120 @@ const char *EC_curve_nid2nist(int nid)
 int EC_curve_nist2nid(const char *name)
 {
     size_t i;
+
     for (i = 0; i < OSSL_NELEM(nist_curves); i++) {
         if (strcmp(nist_curves[i].name, name) == 0)
             return nist_curves[i].nid;
     }
     return NID_undef;
 }
+
+#ifdef FIPS_MODE
+# define MAX_CURVE_ORDER_LEN (571 + 7)/8
+static int ec_curve_nid_from_params(const EC_GROUP *group)
+{
+    int ret = 0, len;
+    size_t i, seed_len;
+    const EC_METHOD *meth = EC_GROUP_method_of(group);
+    int field_type = EC_METHOD_get_field_type(meth);
+    unsigned char ord[MAX_CURVE_ORDER_LEN];
+    const const EC_CURVE_DATA *data;
+    const unsigned char *params_seed, *params_order;
+    EC_GROUP *tmp = NULL;
+    BN_CTX *ctx = NULL;
+    int sz = (EC_GROUP_get_degree(group) + 7) / 8;
```

Comment:
nit: `meth`, `field_type`, `sz` are mixing declarations and code (and assuming `group` and `meth` are non-`NULL`)